### ISSUE TITLE
[BUG] AOT trim warning

### ISSUE THREAD

**Issue by JeyJeyGao** [Azure/azure-sdk-for-net#40892]
### Library name and version

Azure.Core v1.36.0 & Azure.Security.KeyVault.* v4.6.0-beta.2

### Describe the bug

Here is the dependencies for our project:
```
  <ItemGroup>
    <PackageReference Include="Azure.Identity" Version="1.10.4" />
    <PackageReference Include="Azure.Security.KeyVault.Certificates" Version="4.6.0-beta.2" />
    <PackageReference Include="Azure.Security.KeyVault.Keys" Version="4.6.0-beta.2" />
    <PackageReference Include="Azure.Security.KeyVault.Secrets" Version="4.6.0-beta.2" />
    <PackageReference Include="System.Security.Cryptography.Pkcs" Version="8.0.0" />
  </ItemGroup>

```

Encountered dotnet 8 trim warnings: 
```
/home/jj/.nuget/packages/azure.core/1.36.0/lib/net6.0/Azure.Core.dll : warning IL2104: Assembly 'Azure.Core' produced trim warnings. For more information see https://aka.ms/dotnet-illink/libraries [/home/jj/projects/junjie/notation-azure-kv/Notation.Plugin.AzureKeyVault/Notation.Plugin.AzureKeyVault.csproj]
/home/jj/.nuget/packages/azure.security.keyvault.keys/4.6.0-beta.2/lib/netstandard2.0/Azure.Security.KeyVault.Keys.dll : warning IL2104: Assembly 'Azure.Security.KeyVault.Keys' produced trim warnings. For more information see https://aka.ms/dotnet-illink/libraries [/home/jj/projects/junjie/notation-azure-kv/Notation.Plugin.AzureKeyVault/Notation.Plugin.AzureKeyVault.csproj]
```

### Expected behavior

No warnings

### Actual behavior

Encountered dotnet 8 trim warnings

### Reproduction Steps

Build a project with those dependencies:
Here is the dependencies for our project:
```
  <ItemGroup>
    <PackageReference Include="Azure.Identity" Version="1.10.4" />
    <PackageReference Include="Azure.Security.KeyVault.Certificates" Version="4.6.0-beta.2" />
    <PackageReference Include="Azure.Security.KeyVault.Keys" Version="4.6.0-beta.2" />
    <PackageReference Include="Azure.Security.KeyVault.Secrets" Version="4.6.0-beta.2" />
    <PackageReference Include="System.Security.Cryptography.Pkcs" Version="8.0.0" />
  </ItemGroup>

```

Then publish with Donet 8 AOT enabled

### Environment

.NET SDK:
 Version:           8.0.100
 Commit:            57efcf1350
 Workload version:  8.0.100-manifests.6c33ef20

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  22.04
 OS Platform: Linux
 RID:         linux-x64
 Base Path:   /home/jj/download/.dotnet8.0/sdk/8.0.100/

.NET workloads installed:
 Workload version: 8.0.100-manifests.6c33ef20
There are no installed workloads to display.

Host:
  Version:      8.0.0
  Architecture: x64
  Commit:       5535e31a71

.NET SDKs installed:
  8.0.100-preview.3.23178.7 [/home/jj/download/.dotnet8.0/sdk]
  8.0.100-rc.2.23502.2 [/home/jj/download/.dotnet8.0/sdk]
  8.0.100 [/home/jj/download/.dotnet8.0/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 8.0.0-preview.3.23177.8 [/home/jj/download/.dotnet8.0/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.0-rc.2.23480.2 [/home/jj/download/.dotnet8.0/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.0 [/home/jj/download/.dotnet8.0/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 8.0.0-preview.3.23174.8 [/home/jj/download/.dotnet8.0/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.0-rc.2.23479.6 [/home/jj/download/.dotnet8.0/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.0 [/home/jj/download/.dotnet8.0/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  DOTNET_ROOT       [/home/jj/download/.dotnet8.0]

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download

**Comment by jsquire** [comment]
Thank you for your feedback.  Tagging and routing to the team member best able to assist.  Please expect and forgive slow replies during the US holiday season.

**Comment by JeyJeyGao** [comment]
Hi @heaths, could you have a look at the trimming warning of keyvault sdk? Many thanks!

**Comment by heaths** [comment]
A build with trim warnings resolved has not yet been released, but should within a month or so. Already fixed in 65caaa5a1ba32abc5234bd4e7a9fee0102bc8467 and bab8b0a33c094c6389004e7faa96833e7f647e1c.

**Comment by JeyJeyGao** [comment]
Many thanks @heaths !
